base/channels.jl

```diff
@@ -0,0 +1,137 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+abstract AbstractChannel{T}
+
+type Channel{T} <: AbstractChannel{T}
+    cond_take::Condition    # waiting for data to become available
+    cond_put::Condition     # waiting for a writeable slot
+    state::Symbol
+
+    data::Array{T,1}
+    szp1::Int               # current channel size plus one
+    sz_max::Int             # maximum size of channel
+    take_pos::Int           # read position
+    put_pos::Int            # write position
+
+    function Channel(sz)
```

Comment:
You can define a channel size greater than 32. `data` is a circular buffer that is managed via `szp1`, `take_pos` and `put_pos`. The initial allocation is 32, It is grown on demand to a maximum size of `sz_max`. Even a channel of `typemax(Int)` will only allocate storage for 32 objects initially.